### PR TITLE
Add graphical display menu docs

### PR DESCRIPTION
**Pull request by MrMDavidson**
## Description:

Adds documentation for the new Graphical Display Menu.

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#4105

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by MrMDavidson** [comment]
Note: Currently using the same images in the index image table, and documentation, as the `lcd_menu`.

**Review by jesserockz — CHANGES_REQUESTED** [review]
Can you please slim down the configuration example. I had a look at the LCD menu example and its also quite big (not to be changed here) but the main example at the top should be a simple minimum yaml for this component only, and if you want to have more full examples, they should be at the bottom of the page, after all of the yaml actions have also be added.

**Comment by MrMDavidson** [comment]
> Can you please slim down the configuration example. I had a look at the LCD menu example and its also quite big (not to be changed here) but the main example at the top should be a simple minimum yaml for this component only, and if you want to have more full examples, they should be at the bottom of the page, after all of the yaml actions have also be added.

Can do! I need to revisit this PR to add documentation on pop up vs advanced drawing mode in light of the changes in the linked PR. Will do them all together once we're good with the design changes.

File: components/display_menu/graphical_display_menu.rst [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+Overview
+--------
+
+The integration implements the :ref:`Display Menu <display_menu>` integration providing
+a hierarchical menu primarily intended to be controlled either by a rotary encoder
+with a button or a five-button joystick controller.
+
+The component needs to be connected to an instance of a display supporting ESPHome's rendering 
+engine such as :doc:`E-Paper displays <../display/waveshare_epaper>` or :doc:`OLED displays <../display/ssd1306>`.
```
Comment (jesserockz) [inline components/display_menu/graphical_display_menu.rst]: Docs should be absolute, or relative only if in the same directory
```suggestion
engine such as :doc:`E-Paper displays </components/display/waveshare_epaper>` or :doc:`OLED displays </components/display/ssd1306>`.
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

**Review by jesserockz — APPROVED** [review]

**Pull request merged**